### PR TITLE
[cleanup] [test] remove useless TestAuthorizationProvider2

### PR DESCRIPTION
**Pull request by thetumbled**
### Motivation
`TestAuthorizationProvider2` is useless, no need to retain.

### Modifications

Remove it.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Comment by thetumbled** [comment]
PTAL, thanks. @Technoboy- @lhotari

File: pulsar-broker/src/test/java/org/apache/pulsar/client/api/AuthorizationProducerConsumerTest.java [hunk 1]
```diff
@@ -349,7 +349,8 @@ public void testSubscriberPermission() throws Exception {
         } catch (Exception e) {
             // my-sub1 has no msg backlog, so expire message won't be issued on that subscription
             assertTrue(e.getMessage().startsWith("Expire message by timestamp not issued on topic"));
-        }        sub1Admin.topics().peekMessages(topicName, subscriptionName, 1);
+        }
+        sub1Admin.topics().peekMessages(topicName, subscriptionName, 1);
```
Comment (lhotari) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/api/AuthorizationProducerConsumerTest.java:353]: Is this change related to this PR?

Comment (thetumbled) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/api/AuthorizationProducerConsumerTest.java:353]: Just to correct the code format.

**Review by lhotari — APPROVED** [review]
LGTM
